src/include/random.h

```diff
@@ -141,6 +153,11 @@ int generate_random_number(MutexT& m, EngineT& e)
 
 } // namespace detail
 
+bool rng_is_seeded() 
+{
+  return detail::rng_is_seeded_state();
+}
```

Comment:
i would really prefer automatic seeding, similar to what @adamemerson proposed. it seems like every user of this library would be forced into a pattern like this:
```c++
if (!rng_is_seeded()) {
  randomize_rng();
}
auto x = generate_random_number<int>(0, 10);
```